components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/EnvironmentSpecificAPIPropertyDAO.java

```diff
@@ -0,0 +1,284 @@
+        String envsAsList = defaultEnvs.values().stream()
+                .map(env -> "'" + env.getUuid() + "'")
+                .collect(Collectors.joining(","));
+        final String query = getDefaultEnvironmentSpecificAPIPropertiesOfAPIsQuery(apiUuidS, envsAsList);
+        Connection conn = null;
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            try (PreparedStatement preparedStatement = conn.prepareStatement(query);
+                    ResultSet resultSet = preparedStatement.executeQuery()) {
+                while (resultSet.next()) {
+                    String envId = resultSet.getString(1);
+                    String envName = envId; // for default envs envId and envName is same
+                    String apiId = resultSet.getString(2);
+                    JsonObject jsonConfig = null;
+                    InputStream propertyConfigBlob = resultSet.getBinaryStream(3);
+                    if (propertyConfigBlob != null) {
+                        String apiJsonConfig = APIMgtDBUtil.getStringFromInputStream(propertyConfigBlob);
+                        jsonConfig = new Gson().fromJson(apiJsonConfig, JsonObject.class);
+                    }
+                    Map<String, Environment> environmentMap;
+                    Environment environment;
+                    if (apiEnvironmentMap.containsKey(apiId)) {
+                        environmentMap = apiEnvironmentMap.get(apiId);
+                    } else {
+                        environmentMap = new HashMap<>();
+                        apiEnvironmentMap.put(apiId, environmentMap);
+                    }
+                    environment = new Environment();
+                    environment.setEnvId(envId);
+                    environment.setEnvName(envName);
+                    environment.setConfigs(jsonConfig);
+                    environmentMap.put(envName, environment);
+                }
+            }
+        } catch (SQLException e) {
+            throw new APIManagementException("Error occurred when getting default environment specific api properties", e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(null, conn, null);
+        }
+        return apiEnvironmentMap;
+    }
+
+    private String getMgEnvironmentSpecificAPIPropertiesOfAPIsQuery(String apiIds) {
+        return "SELECT AM_GATEWAY_ENVIRONMENT.UUID ENV_ID,"
+                        + "       AM_GATEWAY_ENVIRONMENT.NAME ENV_NAME,"
+                        + "       AM_API.API_UUID API_ID,"
+                        + "       AM_API_ENVIRONMENT_KEYS.PROPERTY_CONFIG CONFIG"
+                        + " FROM AM_API_ENVIRONMENT_KEYS,AM_GATEWAY_ENVIRONMENT,AM_API"
+                        + " WHERE AM_API_ENVIRONMENT_KEYS.ENVIRONMENT_ID = AM_GATEWAY_ENVIRONMENT.UUID AND"
+                        + "        AM_API_ENVIRONMENT_KEYS.API_ID = AM_API.API_ID AND"
+                        + "        AM_API.API_UUID IN (" + apiIds + ")"
```

Comment:
I think we cannot substitute directly as there is a possibility of SQL injections.. Shall we use something like this https://github.com/wso2/carbon-apimgt/blob/389ae2d1ab81a1b4122d73b204d743dae00d9eaf/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java#L5236-L5237